Bluepill-cli/Bluepill-cli/Bluepill/Bluepill.m

```diff
@@ -240,8 +239,10 @@ - (void)createSimulatorWithContext:(BPExecutionContext *)context {
         // If we failed to create the simulator, there's no reason for us to try to delete it, which can just cause more issues
         if (--__self.maxCreateTries > 0) {
```

Comment:
Are you seeing a case where you're hitting create and the simulator already exists? What exactly is the flow that leads to that?
The `if` here basically handles the case where we try to create a simulator and it fails, so we try again `maxCreateTries` times. If it continues to fail, we call `deleteSimulatorWithContext` as a safety measure.
But if it's your first time through the loop, the simulator with the ID used for creation should not exist yet. Deleting prior to creating should always do nothing. What behavior are you seeing?

On @oliverhu's original statement, we added this code because the Travis machines are super low on resources and we couldn't run all of our test cases. By trying and retrying, sometimes resources would free up and we'd be able to create the simulator and continue. If we adopt your change, then it seems we could also get rid of the maxCreateTries and use the global retry as @oliverhu says since it covers that simulator-must-be-deleted-before-we-can-create-it scenario and we no longer care about Travis in particular. But that might be a larger change as there are command line flags exposed to set this value. I'd prefer to see that change in another commit once this one is merged.